### DISCUSSION TITLE
new wezterm windows not focused correctly in KDE

### DISCUSSION THREAD

**Discussion by andreasacwu** [wezterm/wezterm#5870] [Q&A]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

DE: Plasma 6.1.3
WM: KWin

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

hi guys i'm encountering an issue with wezterm and couldn't find a solution after some googling.

when i open a new wezterm window: 
if there are no other wezterm windows open, or 
if there are other wezterm windows open and in focus, 
the new window will be brought to the top correctly. 

however if there are other wezterm windows open and none of the are in focus,
the new window will not come to foreground as i expect it to. 

i wonder why this happens and how i can fix it. thanks :)

**Comment by luerhard** [comment]
This indeed is quite annoying ^^

**Comment by carpii** [comment]
Would love a fix for this too. 

Even trying to force Accept Focus using KDE window rules is ineffective. 

Other terms like Alacritty don't have the same issue, I cant understand whats causing it

**Comment by carpii** [comment]
This workaround works for me (when I open a new wezterm from sxhkd) 

`/usr/bin/wezterm-gui start --always-new-process`

**Reply by clohrmann** [reply]
This is indeed quite annoying. I previously had a keyboard shortcut for "/usr/bin/wezterm" and are facing the same issues.
carpii's workaround seems to do the trick, for now.

**Comment by SinTan1729** [comment]
I'm also facing this, it's a little annoying. Alacritty works properly, so it's probably not a KDE or Wayland limitation.
